roles/kubernetes-apps/csi_driver/cinder/tasks/cinder-credential-check.yml

```diff
@@ -4,15 +4,38 @@
     msg: "cinder_auth_url is missing"
   when: cinder_auth_url is not defined or not cinder_auth_url
 
-- name: Cinder CSI Driver | check cinder_username value
+- name: Cinder CSI Driver | check cinder_username value cinder_application_credential_name value
   fail:
-    msg: "cinder_username is missing"
-  when: cinder_username is not defined or not cinder_username
+    msg: "you must either set cinder_username or cinder_application_credential_name"
+  when:
+    - cinder_username is not defined or not cinder_username
+    - cinder_application_credential_name is not defined or not cinder_application_credential_name
+
+- name: Cinder CSI Driver | check cinder_application_credential_id value
+  fail:
+    msg: "cinder_application_credential_id is missing"
+  when:
+    - cinder_application_credential_name is defined
+    - cinder_application_credential_name|length > 0
+    - cinder_application_credential_id is not defined or not cinder_application_credential_id
+
+- name: Cinder CSI Driver | check cinder_application_credential_secret value
+  fail:
+    msg: "cinder_application_credential_secret is missing"
+  when:
+    - cinder_application_credential_name is defined
+    - cinder_application_credential_name|length > 0
+    - cinder_application_credential_secret is not defined or not cinder_application_credential_secret
 
 - name: Cinder CSI Driver | check cinder_password value
   fail:
     msg: "cinder_password is missing"
-  when: cinder_password is not defined or not cinder_password
+  when:
+    - cinder_username is defined
+    - cinder_username|length > 0
+    - cinder_application_credential_name is not defined or not cinder_application_credential_name
+    - cinder_application_credential_secret is not defined or not cinder_application_credential_secret
```

Comment:
nit: This cinder_application_credential_secret is checked at the task of line 22.
I guess we can remove this line 37 for checking cinder_password.